src/plugins/index_pattern_field_editor/public/types.ts

```diff
@@ -43,26 +43,17 @@ export interface StartPlugins {
 
 export type InternalFieldType = 'concrete' | 'runtime';
 
-export interface Field {
+export interface Field extends EnhancedRuntimeField {
   name: string;
-  type: RuntimeField['type'] | string;
-  script?: RuntimeField['script'];
-  customLabel?: string;
-  popularity?: number;
-  format?: FieldFormatConfig;
 }
 
-export interface FieldFormatConfig {
-  id: string;
-  params?: { [key: string]: any };
+export interface CompositeField extends RuntimeField {
+  type: 'composite';
+  name: string;
+  subFields: Record<string, EnhancedRuntimeField>;
 }
 
-export interface EsRuntimeField {
-  type: RuntimeType | string;
-  script?: {
-    source: string;
-  };
-}
+export type FieldFormatConfig = EnhancedRuntimeField['format'];
```

Comment:
I don't think this is public as it is not exported from `public/index.ts`.

Reply:
Ah, gotcha. That might not mean it _shouldn't_ be public, however.  I'm trying to see if it's used in a public type and there are quite a few types that are not exported from this plugin that should be (see https://docs.elastic.dev/kibana-dev-docs/best-practices#export-every-type-used-in-a-public-api for more info).

To explain this a bit more, see how in our docs [here](https://docs.elastic.dev/kibana-dev-docs/indexPatternFieldEditorPluginApi#if7d77641-0996-11ec-9277-c5da131bc24d), `FormatEditorState` isn't clickable?

<img width="851" alt="Screen Shot 2021-08-30 at 10 04 39 AM" src="https://user-images.githubusercontent.com/16563603/131351897-efd2bb65-74e9-4158-aeaa-690c4f7fa2c8.png">

That's because it should be exported but it's not. If you run ` node scripts/build_api_docs.js --plugin indexPatternFieldEditor --stats exports` it'll tell you the types that should be exported but aren't.

Once you start exporting these types, it proliferates into a much larger public API than you might realize.

E.g. FormatEditorState uses `SampleInput`, `Sample` and `SampleInputs` - all types that aren't actually exported.

Since this type already exists, I think this is fine to leave as is in this PR, but there are other new types that need to be exported. You can see this in the doc output:

 > // Warning: (ae-forgotten-export) The symbol "RuntimeCompositeWithSubFields" needs to be exported by the entry point index.d.ts
 
 A bunch of those will need to be exported.  Those legacy docs have been deleted but there are ci stats that will show up if you increased the number.

Reply:
It makes sense, thanks for the detailed info 👍 

I removed the alias and used the `SerializedFieldFormat` directly ([here](https://github.com/elastic/kibana/pull/110226/commits/2465b15ffb221c131db0b81732fe9d6ab707aa37) and [here](https://github.com/elastic/kibana/pull/110226/commits/4894c294571801c14bb02394d3128308520214e0)). 
I have also exported all the publicly accessible interfaces as you suggest ([here](https://github.com/elastic/kibana/pull/110226/commits/d246459dacc7017528bcaf0a90be17613dc48305))